src/domain/signup/signupServerErrorsContext/utils.ts

```diff
@@ -45,6 +46,15 @@ export const parseSignupGroupServerErrors = ({
     error: LEServerError;
     key: string;
   }) {
+    if (
```

Comment:
Otherwise this PR looks good but some of the newly-added conditions in this file are overflowing the cognitive complexity limit. In my opinion the code is still understandable and not too complex, but perhaps consider refactoring this a bit to try and prevent snowballing complexity later.

Reply:
My personal opinion is that it's not too complex. But of course we can make it little bit easier to understand. I did some refactoring to reduce cognitive complexity.